app/cdap/components/RecipeManagement/CreateRecipe/index.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { FormEvent, ChangeEvent, useEffect, useState, useRef, useCallback } from 'react';
+import T from 'i18n-react';
+import { getRecipeByName, createRecipe } from 'components/RecipeManagement/CreateRecipe/services';
+import { debounce } from 'lodash';
+import { IRecipeFormData } from 'components/RecipeManagement/types';
+import RecipeForm, { CREATE_RECIPE } from 'components/RecipeManagement/RecipeForm';
+import { ICreateRecipeProps } from 'components/RecipeManagement/types';
+
+const PREFIX = 'features.WranglerNewUI.RecipeForm.labels';
+const recipeNameRegEx = /^[a-z\d\s]+$/i;
+
+const noErrorState = {
+  isRecipeNameError: false,
+  recipeNameErrorMessage: '',
+};
+
+export default function CreateRecipe({ setShowRecipeForm, setSnackbar }: ICreateRecipeProps) {
+  const [recipeFormData, setRecipeFormData] = useState<IRecipeFormData>({
+    recipeName: '',
+    description: '',
+    directives: [],
+  });
+
+  const [isSaveDisabled, setIsSaveDisabled] = useState<boolean>(true);
+  const [recipeNameErrorData, setRecipeNameErrorData] = useState(noErrorState);
+
+  // This static data has to be removed when we have actual API data, then directly we will get that data from store as directives
+  const recipeSteps = ['uppercase: body1', 'titlecase: body2'];
+
+  useEffect(() => {
+    handleSaveButtonMode();
+  }, [recipeNameErrorData.isRecipeNameError]);
```

Comment:
This `useEffect` can be removed; please check the comment above.

Reply:
Removed UseEffect